### DISCUSSION TITLE
Pyright not handles copying TypedDict

### DISCUSSION THREAD

**Discussion by kumaraditya303** [microsoft/pyright#2597] [Ideas]
See this code:

```py
# pyright: strict
from typing import TypedDict


class A(TypedDict):
    name: str


a = A(name="foo")
b = {**a} # Make a copy to not mutate a
reveal_type(b)
```

Revealed Type should be A not `dict[str, object]`. Can support for copying be added ?

**Comment by erictraut** [comment]
This is the expected behavior. It's consistent with other type checkers. See https://github.com/python/mypy/issues/2738 for a related discussion.

You can use `b = A(**a)` if you would like to construct a new TypedDict.

**Reply by kumaraditya303** [reply]
The issue is that at runtime A cannot be imported so `b = A(**a)` is not possible. Current workaround `a.copy()`
